_episodes/l4_04_parameterized_tests.md

```diff
@@ -0,0 +1,340 @@
+---
+title: "Parameterised Tests using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What is a parameterised test?"
+- "How can I write my own parameterised tests using GoogleTest?"
+objectives:
+- "Understand why do we need a parameterised test."
+- "Understand how to create a parameterised test using GoogleTest."
+- "Understand the advantages of parameterised tests."
+- "Understand how to combine a test fixtures with parameterised test."
+keypoints:
+- "We learnt about parameterised tests and how to create them in GoogleTest."
+- "We learnt how parameterised tests are better as compared to using a loop for testing multiple values, conditions, branches etc."
+- "We learnt how to combine test fixture with parameterised test."
```

Comment:
Like in the fixtures lessons, this should be take away messages. I'd suggest the following:

```suggestion
- "Parameterised tests can be used to repeat a specific test with different inputs, reducing code duplication."
- "Parameterised tests are individual tests, so they are more concise and easy to maintain than using a loop for testing multiple conditions"
- "Fixtures can be combined with parameterised tests for maximum flexibility."
```